offline/QA/Jet/photonjetskinematics.cc

```diff
@@ -0,0 +1,184 @@
+#include <calobase/TowerInfoContainer.h>
+#include <calobase/TowerInfoDefs.h>
+
+// Cluster includes
+#include <calobase/RawCluster.h>
+#include <calobase/RawClusterContainer.h>
+
+
+//____________________________________________________________________________..
+photonjetskinematics::photonjetskinematics(const std::string &outputfilename):
+SubsysReco("photonjetskinematics")
+, outfilename(outputfilename)
+{
+ std::cout << "photonjetskinematics::photonjetskinematics(const std::string &name, const std::string&outputfilename) Calling ctor" << std::endl;
+}
+//____________________________________________________________________________..
+photonjetskinematics::~photonjetskinematics()
+{
+  std::cout << "photonjetskinematics::~photonjetskinematics() Calling dtor" << std::endl;
+}
+
+//____________________________________________________________________________..
+int photonjetskinematics::Init(PHCompositeNode *topNode)
+{
+  std::cout << "photonjetskinematics::Init(PHCompositeNode *topNode) Initializing" << std::endl;
+  //outfile = new TFile(outfilename.c_str(), "RECREATE");
+  PHTFileServer::get().open(outfilename, "RECREATE");
+
+  //initializing histograms
+
+  h_emcal_cluster_chi2 = new TH1D("h_emcal_cluster_chi2", "h_emcal_cluster_chi2", 30, 0, 150); //1000,0,5e6
+  h_emcal_cluster_chi2->GetXaxis()->SetTitle("#chi^{2}");
+ 
+
+  h_emcal_cluster_energy = new TH1D("h_emcal_cluster_energy", "h_emcal_cluster_energy", 100, 0, 10);
+  h_emcal_cluster_energy->GetXaxis()->SetTitle("E_{T} [GeV]");
+
+
+  h_emcal_cluster_eta_phi = new TH2F("h_emcal_cluster_eta_phi", "h_emcal_cluster_eta_phi", 140, -1.2, 1.2, 64, -M_PI, M_PI);
+  //eta used to be 24 increase to 100
+  //TH2D changed to TH2F 
+  h_emcal_cluster_eta_phi->GetXaxis()->SetTitle("#eta");
+  h_emcal_cluster_eta_phi->GetYaxis()->SetTitle("#Phi");
+  h_emcal_cluster_eta_phi->SetOption("COLZ");
+
+
+  h_emcal_cluster_eta = new TH1D("h_emcal_cluster_eta", "Eta Distribution", 140, -1.2, 1.2);
+  h_emcal_cluster_eta->GetXaxis()->SetTitle("#eta");
+
+  h_emcal_cluster_phi = new TH1D("h_emcal_cluster_phi", "Phi Distribution", 64, -M_PI, M_PI);
+  h_emcal_cluster_phi->GetXaxis()->SetTitle("#phi");
+
+  return Fun4AllReturnCodes::EVENT_OK;
+  
+}
+
+//____________________________________________________________________________..
+int photonjetskinematics::InitRun(PHCompositeNode *topNode)
+{
+  std::cout << "photonjetskinematics::InitRun(PHCompositeNode *topNode) Initializing for Run XXX" << std::endl;
+  return Fun4AllReturnCodes::EVENT_OK;
+}
+
+//____________________________________________________________________________..
+int photonjetskinematics::process_event(PHCompositeNode *topNode)
+{  
+
+  RawClusterContainer* clusterContainer = findNode::getClass<RawClusterContainer>(topNode, "CLUSTERINFO_CEMC");
+  if (!clusterContainer)
+    {
+      std::cout << PHWHERE << "funkyCaloStuff::process_event - Fatal Error - CLUSTER_CEMC node is missing. " << std::endl;
+      return 0;
+    }
+
```

Comment:
We'll need to add a chunk of code here to filter out events which don't have the trigger we're interested in if we're applying a trigger selection.
```suggestion
// if needed, check if trigger fired
if (doTrgSelect)
{
  bool hasTrigger = JetQADefs::DidTriggerFire(trgToSelect, topNode);
  if (!hasTrigger)
  {
    return Fun4AllCodes::EVENT_OK;
  }
}

```